components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -2514,3 +2540,323 @@ def object_table_query(request, objtype, objid, conn=None, **kwargs):
     tableData['parentId'] = ann['parentId']
     tableData['addedOn'] = ann['addedOn']
     return tableData
+
+
+def build_url(request, name, api_version, **kwargs):
```

Comment:
Just discussing this with @aleksandra-tarkowska. One option is to allow configuration. By default we use ```build_absolute_url()``` but you could configure it to use just the relative part of the url OR to use a base url of your choosing. Seems that ```build_absolute_url()``` issues are only related to caching, right? Caching is really useful for public repositories, but won't be used by "regular" uses so much.